### ISSUE TITLE
Add in EU Cookie Consent

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#866]
Allow Geeklog site to display a Cookie Consent. 

Either add to themes or bake into Geeklog.

Maybe use code from here:

https://cookieconsent.insites.com/

Have option  to enable or disable cookie consent.

**Comment by eSilverStrike** [comment]
Added 2c2141ccf0214dd7f64eb91d568df867084f6c16 to better support this feature giving themes the ability to set a block of javascript code using the scripts class

**Comment by mystralkk** [comment]
Added with change set cc7e861.  However, I cannot fully check if this feature is working properly, since I live in Japan.

**Comment by eSilverStrike** [comment]
Working on the Geeklog.net  theme and want to add the ability to customize easier based on theme.
I will also add 3 new config fields for: 

- terms of use
- privacy policy
- about cookies

**Comment by mystralkk** [comment]
If you would like to customize the cookie consent, visit https://cookieconsent.osano.com/download/ .  They seem to have set up a firm and things are getting a bit complicated.

**Comment by eSilverStrike** [comment]
I have the customizations working fine for the cookie consent. 

I am trying to add the config values (they are links to those pages) so they support multiple languages. (ie you can have a link for en, and a different link for ja). Unfortantly the config array support in the Geeklog Configuration isn't quite there. I have it 90% working at the moment. If I can't get it right in the next little bit I will just use normal text fields for the new config options and then add a new feature request for these config fileds in Geeklog 2.2.2. 

The Multilanguage support is taking a little longer than expected and I want to get Beta 2 out soon.

**Comment by eSilverStrike** [comment]
Okay I did not add the multilanguage support for the new config options I added. I figured we should do it for all language based config options  at the same time. For more information see #990 for the improvement request.

With last commit I added:
- Added the ability for Cookie Consent to include customizations. 
- Added new config options about_cookies_link, terms_of_use_link, privacy_policy_link. This options support multiple languages via COM_switchLocaleSettings
- Added ability for Geeklog Configuration to validate text arrays that include custom elements and display the error. Also include example of custom validation functions for this 
- index.thtml template will has new template variables for about_cookies_link, terms_of_use_link, privacy_policy_link along with corresponding language variables
